### ISSUE TITLE
OpenAPIReference does not work for SDK generators that create C# output in a folder

### ISSUE THREAD

**Issue by darrelmiller** [dotnet/aspnetcore#38547]
### Describe the bug
The following build action is executed when you use  a SDK generator that uses OutputPath to place the generated C# files.

```xml
      <Compile Include="@(_Directories -> '%(Identity)/**/*.$(DefaultLanguageSourceExtension)')"
          Exclude="@(Compile)"
          Condition="'$(DefaultLanguageSourceExtension)' != '.ts'">
        <SourceDocument>%(_Directories.FullPath)</SourceDocument>
      </Compile>
```
https://github.com/dotnet/aspnetcore/blob/64c37113aafcbad73113b7cb86f86129e8839963/src/Tools/Extensions.ApiDescription.Client/src/build/Microsoft.Extensions.ApiDescription.Client.targets#L129

This outputs a CSC command that includes the globbed path as a parameter.  From my tests CSC doesn't appear to support path globbing.

### To Reproduce
Add an OutputPath value to the OpenAPIReference in the csproj file.  Unfortunately I think NSwag fails with this setting so it isn't as easy to see the final error.  You first need to fix the NSwag .targets file.  I can provide full steps to repro using kiota as a SDK generator if needed.

### Exceptions (if any)

This produces the following error at build time.

```
CSC : error CS2021: File name 'obj\sdk/**/*..cs' is empty, contains invalid characters, has a drive specification without an absolute path, or is too long
```
I think the extra dot is a problem too, but even after removing that, the globbing is still a problem.

### Further technical details

I tried this on .net 5 and 6.  Same error.

**Comment by dougbu** [comment]
> I think the extra dot is a problem too, but even after removing that, the globbing is still a problem.

The globbing _might_ be an issue but the dot definitely is. Could you please try removing that period and report the new error (if any)❔

**Comment by darrelmiller** [comment]
@dougbu Removing the dot produces the same error from CSC.

**Comment by dougbu** [comment]
> Removing the dot produces the same error from CSC.

Is the error now `CSC : error CS2021: File name 'obj\sdk/**/*.cs' is empty`❔ I ask because something is weird if the `Include` isn't populating the `@(Compile)` item group as expected.

I could probably use a binary log from your test failure because I don't have a generator that creates directories on my machine.

**Comment by darrelmiller** [comment]
This is what I get when I remove the dot from the compile action. 

![image](https://user-images.githubusercontent.com/447694/142928790-58802ffe-7775-4a5a-8e87-c7c5ca394d36.png)

**Comment by darrelmiller** [comment]
The generator tool isn't yet in a public feed, so you have to jump through a couple of hoops to install the generator.  https://github.com/microsoft/kiota/blob/main/docs/generator/tool.md

**Comment by dougbu** [comment]
Something strange is going on because item groups support globbing, including the `**` part. I'll dive deeper later today and see what I'm missing&hellip;

**Comment by dougbu** [comment]
The root cause here is dotnet/msbuild#3274. Globbing inside an item transform apparently doesn't work. /fyi @rainersigwald

I have a workaround and will put up a PR to fix things for 7.0.0-preview1 in a bit. Is testing w/ our nightly builds sufficient for now❔

**Comment by darrelmiller** [comment]
@dougbu I can definitely test with the nightly build.

**Comment by baywet** [comment]
@dougbu thanks for working on this! If I understand the release process correctly, once the PR is merged, it'll go out with the next minor/patch public release of dotnet 6, correct? and then this will get picked up in the next quarterly update of visual studio too?

**Comment by dougbu** [comment]
@baywet not quite. If we merge #39293 soon, the fix should be part of 6.0.2. And, since the affected code ships only in the Microsoft.Extensions.ApiDescription.Client package, nothing relevant here will be visible in VS except the new package version.

**Comment by baywet** [comment]
thanks for the additional information. So the only requirement on VS users to see the fix would be to upgrade their dependency then? even better!
